### PR TITLE
Add attribute collecting to standard shader chunk

### PR DESCRIPTION
**Pull request by Mikecoon**
This pull request changes standard program lib in order to make in collect custom attributes from VertexShader by pc.shaderChunks.collectAttribs.

Afterwards, merges it with existing attributes.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,5 +38,8 @@
   },
   "engines": {
     "node": ">= 0.6.12"
+  },
+  "dependencies": {
+    "npm": "^5.8.0"
```
Comment (willeastcott) [inline package.json]: Did you mean to add this dependency? If so, why is it needed in this PR?

**Comment by willeastcott** [comment]
Hey @mvaligursky - this PR has been open for 2 years! Wow. Can you take a look and make a final judgement with what to do with it please?

**Comment by mvaligursky** [comment]
this cannot be added directly. As an example, it breaks instancing (engine/examples/#graphics/hardware-instancing.html).

**Reason:**

in standard.js, instancing is mapped to textcoords
https://github.com/playcanvas/engine/blob/f591bbeeb6ae6151f04f4e5ef0c11fcb1093e4b7/src/graphics/program-lib/standard.js#L546-L552

but in chunks.js, instancing is not part of attrib2Semantic
https://github.com/playcanvas/engine/blob/f591bbeeb6ae6151f04f4e5ef0c11fcb1093e4b7/src/graphics/chunks.js#L13-L28

and so it would add additional attribute "ATTR0". And that's not correct.
https://github.com/playcanvas/engine/blob/f591bbeeb6ae6151f04f4e5ef0c11fcb1093e4b7/src/graphics/chunks.js#L45

I think in ideal situation, we remove manual processing of attributes in standard.js, and only use the line this PR adds, similarly to how collectAttribs is used in particle.js - it just collects attributes, but does not add to attributes opening double adding issue.

**Comment by willeastcott** [comment]
@mvaligursky Since this PR is quite old, @Mikecoon may no longer be available to respond to your feedback (but let us know if you are!). If we don't hear anything, you may have to attack this yourself, or alternatively just close it. Let's give @Mikecoon a few days to respond though.

**Comment by Mikecoon** [comment]
I'm here, but I don't quite follow what should I change to address it.
And seems like I removed this repo :c

**Comment by mvaligursky** [comment]
I've decided to close this. It's a good idea, and I wish it would work, but our shader chunks contain more attributes than the final shader needs.

Example - looking at character shader that gets generated for one of the engine examples, and the chunk contains "attribute vec4 vertex_tangent;" and that line is added to vscode. But it is not used at all in that shader, as code that uses it is added conditionally based on some conditions is not included. In this shader, there's about 4 attributes that are in vshader code but not used by it.

So we cannot scan vshader and use all attributes from it.

**Pull request closed without merging**